--- src/mediascan_unix.c	2013-03-10 13:38:24.791826689 +0100
+++ src/mediascan_unix.c	2013-03-10 14:37:40.437589298 +0100
@@ -9,6 +9,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <unistd.h>
+#include <sys/stat.h>
 
 #include <libmediascan.h>
 #include "common.h"
@@ -103,6 +104,7 @@ void recurse_dir(MediaScan *s, const cha
 
   while ((dp = readdir(dirp)) != NULL) {
     char *name = dp->d_name;
+    struct stat st;
 
     // skip all dot files
     if (name[0] != '.') {
@@ -111,7 +113,8 @@ void recurse_dir(MediaScan *s, const cha
         break;
 
       // XXX some platforms may be missing d_type/DT_DIR
-      if (dp->d_type == DT_DIR) {
+      //if (dp->d_type == DT_DIR) {
+      if ((stat(name, &st) == 0) && S_ISDIR(st.st_mode)) {
         // Add to list of subdirectories we need to recurse into
         struct dirq_entry *subdir_entry = malloc(sizeof(struct dirq_entry));
 
